
# HG changeset patch
# User Todd C. Miller <Todd.Miller@sudo.ws>
# Date 1606767492 25200
# Node ID 41db1aad85bbce444b511bcb28b9628a0a22fcd9
# Parent  96a5cfe3c66b6ee84d7f7b6fb26d932f45a44b76
Fix build when configured using --without-sendmail
Bug #947

diff -r 96a5cfe3c66b -r 41db1aad85bb lib/eventlog/eventlog.c
--- a/lib/eventlog/eventlog.c	Sun Nov 29 15:11:34 2020 -0700
+++ b/lib/eventlog/eventlog.c	Mon Nov 30 13:18:12 2020 -0700
@@ -78,7 +78,7 @@
 static FILE *eventlog_stub_open_log(int type, const char *logfile);
 static void eventlog_stub_close_log(int type, FILE *fp);
 
-/* Eventlog config settings */
+/* Eventlog config settings (default values). */
 static struct eventlog_config evl_conf = {
     EVLOG_NONE,			/* type */
     EVLOG_SUDO,			/* format */
@@ -91,7 +91,11 @@
     false,			/* omit_hostname */
     _PATH_SUDO_LOGFILE,		/* logpath */
     "%h %e %T",			/* time_fmt */
+#ifdef _PATH_SUDO_SENDMAIL
     _PATH_SUDO_SENDMAIL,	/* mailerpath */
+#else
+    NULL,			/* mailerpath (disabled) */
+#endif
     "-t",			/* mailerflags */
     NULL,			/* mailfrom */
     MAILTO,			/* mailto */
@@ -1436,8 +1440,10 @@
 	evl_conf.logpath = _PATH_SUDO_LOGFILE;
     if (evl_conf.time_fmt == NULL)
 	evl_conf.time_fmt = "%h %e %T";
+#ifdef _PATH_SUDO_SENDMAIL
     if (evl_conf.mailerpath == NULL)
 	evl_conf.mailerpath = _PATH_SUDO_SENDMAIL;
+#endif
     if (evl_conf.mailerflags == NULL)
 	evl_conf.mailerflags = "-t";
     if (evl_conf.mailto == NULL)

